reana_commons/consumer.py

```diff
@@ -21,51 +21,50 @@
 # submit itself to any jurisdiction.
 """REANA-Commons module to manage AMQP consuming on REANA."""
 
-import pika
+from kombu import Connection, Exchange, Queue
+from kombu.mixins import ConsumerMixin
 
-from .config import BROKER_PASS, BROKER_PORT, BROKER_URL, BROKER_USER
+from reana_commons.config import (BROKER, MQ_DEFAULT_EXCHANGE,
+                                  MQ_DEFAULT_QUEUE, MQ_DEFAULT_ROUTING_KEY,
+                                  MQ_DEFAULT_SERIALIZER)
 
 
-class Consumer:
-    """Base MQ Consumer."""
+class REANABaseConsumer(ConsumerMixin):
+    """Base RabbitMQ consumer."""
 
-    def __init__(self, queue):
-        """Constructor."""
-        self.broker_credentials = pika.PlainCredentials(BROKER_USER,
-                                                        BROKER_PASS)
-        self._params = pika.connection.ConnectionParameters(
-            BROKER_URL, BROKER_PORT, '/', self.broker_credentials)
-        self._conn = None
-        self._channel = None
-        self._queue = queue
+    def __init__(self, connection=None, queues=None, default_serializer=None):
+        """Construct a REANAConsumer.
 
-    def connect(self):
-        """Connect to MQ channel."""
-        if not self._conn or self._conn.is_closed:
-            self._conn = pika.BlockingConnection(self._params)
-            self._channel = self._conn.channel()
-            self._channel.basic_qos(prefetch_count=1)
-            self._channel.queue_declare(queue=self._queue)
+        :param connection: kombu.Connection object.
+        :param queues: List of queues names the consumer will subscribe to.
+        """
+        self.queues = queues or self._build_default_queues()
+        self.connection = connection or Connection(BROKER)
+        self.default_serializer = default_serializer or MQ_DEFAULT_SERIALIZER
```

Comment:
Maybe `serialization_format` or `message_serialization_format`? Default could refer to a single option that this field could take, so the config variable makes sense.